### ISSUE TITLE
DEVID EN LOKAKEY

### ISSUE THREAD

**Issue by Jelmer-Meurisse** [wvissers/openhab2-addons-tuya#1]
where did you find de DEVID EN LOKALKEY of de LSC??

**Comment by wvissers** [comment]
Two methods I found online are  [instructions using Windows](http://www.htgsd.com/information-technology/apple/homekit/how-to-capture-tuya-lan-homebridge-device-devid-and-key-on-windows-10/) and [setup instructions](https://github.com/codetheweb/tuyapi/blob/master/docs/SETUP.md) to find the devId and localKey values. It requires a setup on e.g. a laptop to grab the data from the app (I use the Tuya app, but the LSC app probably works also).

**Comment by Jelmer-Meurisse** [comment]
oke thank you
so you run a program on the pc and then you can get het 2 ids?

**Comment by wvissers** [comment]
Yes, you use the pcc as a proxy to intercept the traffic between the app on your phone and the Tuya cloud service. It will take a while to figure it out but you only have to do it once. When you have the traffic, look at the messages and there should be some JSON like "devId": "xxxxx", and "localKey": "yyyyy" in it. 
Unfortunately I haven't found a more user friendly way to get this info. If someone knowns a better way, please comment on this issue.

**Comment by Jelmer-Meurisse** [comment]
i can find the devid but the localkey not

**Comment by Jelmer-Meurisse** [comment]
![{498AC15A-8DE3-420F-85A8-822ED926B5E6} png](https://user-images.githubusercontent.com/43674164/64199437-9dd06600-ce8a-11e9-96d5-1f0dadff18dd.jpg)

**Comment by wvissers** [comment]
I think this is the wrong message. You are seeing the post data here, which is actually the request sent to the cloud. You should look for the response instead. 
BTW, have you tried to refresh the device list in the app by sliding it down? Then you should see a couple of requests. There is a rather large response containing a lot of detailed information. Searching for "localKey" in the json data should get you started.

**Comment by Jelmer-Meurisse** [comment]
![Aantekening 2019-09-04 110302](https://user-images.githubusercontent.com/43674164/64241254-9dc07c80-cf03-11e9-80b6-7e74aa9b0b12.jpg)

**Comment by Jelmer-Meurisse** [comment]
how can i see wich device it is

**Comment by wvissers** [comment]
You can look for the "name". If you named the devices in the app, you should be able to find that name in the json data as well.

**Comment by Jelmer-Meurisse** [comment]
oke but is the data after the name?
key en id?

**Comment by Jelmer-Meurisse** [comment]
do you have teamviewer?

**Comment by Jelmer-Meurisse** [comment]
teamvieuwer id 
1 245 121 320

**Comment by wvissers** [comment]
I don't have teamviewer but I'm behind a company firewall here anyway. If I look at the json data, the order of the relevant data seems to be "devId", "name", "localKey".

**Comment by Jelmer-Meurisse** [comment]
![Aantekening 2019-09-04 112828](https://user-images.githubusercontent.com/43674164/64243149-27be1480-cf07-11e9-85d8-c98a1d61de84.jpg)

**Comment by wvissers** [comment]
This status is the normal initial status. It discovered the ip address of the device. That means the devId is recognized. After about 10 seconds that should change to the online status if everything is working fine. 
If the device stays offline, there may be a API version mismatch (currently only API version 3.3 supported) or a mismatch in the local key.

**Comment by Jelmer-Meurisse** [comment]
oke i have this 
 java.net.SocketException

**Comment by Jelmer-Meurisse** [comment]
and how did you know which api hou have?

**Comment by wvissers** [comment]
I have seen Socket Exceptions at my end also in some occasions. I haven't figured out exactly when this happens. In my case it helped restarting openHAB. But this certainly needs to be improved. I will look at it after my holiday 4 weeks from now.
The version should be visible when you click on the PROPERTIES in the paper UI. It should show the ip address and API version.
